torch_xla/csrc/pooling.cpp

```diff
@@ -348,16 +349,27 @@ xla::XlaOp ComputeMaxPoolIndices(
 
 }  // namespace
 
-bool IsSupportedAdaptiveAvgPool2d(absl::Span<const xla::int64> input_size,
-                                  absl::Span<const xla::int64> output_size) {
+bool IsSupportedAdaptiveAvgPool(absl::Span<const xla::int64> input_size,
+                                  absl::Span<const xla::int64> output_size,
+                                  int unsigned pool_dim) {
+
   xla::int64 rank = input_size.size();
-  for (int spatial_dim = 0; spatial_dim < 2; ++spatial_dim) {
-    if (input_size[rank - 2 + spatial_dim] % output_size[spatial_dim] != 0) {
+  for (int spatial_dim = 0; spatial_dim < pool_dim; ++spatial_dim) {
+    if (input_size[rank - pool_dim + spatial_dim] % output_size[spatial_dim] != 0) {
       return false;
     }
   }
   return true;
 }
+bool IsSupportedAdaptiveAvgPool3d(absl::Span<const xla::int64> input_size,
```

Comment:
You might as well just call `IsSupportedAdaptiveAvgPool` instead of creating this function, same for 2d.